lib/view/monthlyTaskView.dart

```diff
@@ -1,60 +1,76 @@
 import 'package:flutter/material.dart';
+import 'package:get/get.dart';
 import 'package:planner/common/database.dart';
-import 'package:planner/common/time_management.dart';
 import 'package:planner/models/task.dart';
 import 'package:table_calendar/table_calendar.dart';
 import 'package:planner/view/taskView.dart';
+import 'package:planner/common/time_management.dart';
 
 class MonthlyTaskView extends StatefulWidget {
-  const MonthlyTaskView({super.key});
-
   @override
   _MonthlyTaskViewState createState() => _MonthlyTaskViewState();
 }
 
 class _MonthlyTaskViewState extends State<MonthlyTaskView> {
-  CalendarFormat _calendarFormat = CalendarFormat.month;
+  final CalendarFormat _calendarFormat = CalendarFormat.month;
   DateTime _focusedDay = DateTime.now();
   DateTime? _selectedDay;
   DatabaseService db = DatabaseService();
   List<Task> todayTasks = [];
-
-  // Add a PageController for handling page navigation
-   final PageController _pageController = PageController();
+  final PageController _pageController = PageController();
+  Map<DateTime, List<Task>> active = {};
 
   @override
   void initState() {
     super.initState();
-    // Add a listener to the PageController to update the focusedDay
-    _pageController.addListener(() {
-      setState(() {
-        _focusedDay = _pageController.page == 0
-            ? _focusedDay = getDateOnly(_focusedDay, offsetMonths: -1)
-            : _focusedDay = getDateOnly(_focusedDay, offsetMonths: 1);
-      });
-    });
+    fetchMonthlyTasks(DateTime.now());
   }
 
   @override
   void dispose() {
-    // Dispose of the PageController to prevent memory leaks
     _pageController.dispose();
     super.dispose();
   }
 
+// Use this function when you're setting and getting tasks from the active map
+  void fetchMonthlyTasks(DateTime selectedDate) async {
+    DateTime dateStart = getDateOnly(selectedDate);
+    Map<DateTime, List<Task>> activeMap, delayedMap, completedMap;
+    (activeMap, delayedMap, completedMap) =
+        await db.getTaskMapsMonth(dateStart);
+
+    active = activeMap.map((key, value) => MapEntry(
+        getDateOnly(key), value)); // Use getDateOnly when setting tasks in the active map
+    todayTasks = active[getDateOnly(selectedDate)] ??
+        []; // Use getDateOnly when getting tasks from the active map
+
+    setState(() {});
+    print(todayTasks);
+  }
```

Comment:
what does this do?

Reply:
From my understanding it collects the set of 3 tasks into a single list